packages/squiggle-lang/__tests__/ReducerProject/ReducerProject_tutorial_5_calling_functions.res

```diff
@@ -0,0 +1,39 @@
+@@warning("-44")
+module InternalExpressionValue = ReducerInterface_InternalExpressionValue
+module Project = ForTS_ReducerProject
+module Bindings = Reducer_Bindings
+
+open Jest
+open Expect
+open Expect.Operators
+
+describe("ReducerProject Tutorial", () => {
+  /* Let's build a project to provide a function. */
+  /* But we will call that function on an array of user input. */
+  let project = Project.createProject()
+  Project.setSource(project, "library", "double(x) = x * 2")
+  /* userCode is not here yet but its dependency is fixed. So we can set it once and for all */
+  Project.setContinues(project, "userCode", ["library"])
+
+  let userValues = [1, 2, 3, 4, 5]
+
+  let userResults = Belt.Array.map(userValues, aUserValue => {
+    let userCode = `double(${aUserValue->Js.Int.toString})`
```

Comment:
In most cases at this point, I assume we won't regularly know that the function is named ``double``. Is there any easy way to inspect this, or something?

Reply:
I'm also curious how we could render anonymous functions.

Reply:
> I assume we won't regularly know that the function is named double. Is there any easy way to inspect this, or something?

We discussed this with Umur yesterday!

One possible approach is to always keep track of the function's name (really, full path, it could be `foo.bar[5].baz(...)`) and we might keep track of Project object as well. Uh, and the source file the function came from.

Then, SquiggleLambda TS objects could wrap code generation and have a simple external API, `lambda.call(new SquiggleNumber(123))` (and, with autoboxing, it could even be `lambda.call(123)`.

That would violate the convention of "TS wrapper classes are just a thin layer on top of Rescript", but in this specific case maybe that's ok.

> I'm also curious how we could render anonymous functions.

I'm not sure I understand this — anonymous functions are still stored in a named variable.

I believe internally squiggle doesn't make a distinguish between `f(x) = x*2` and `f = {|x|x*2}`.

Reply:
That sounds good to me. Seems like a bunch of things to track, but maybe not that bad. 

Agreed on having a nice TS wrapper.